src/ast/CMakeLists.txt

```diff
@@ -100,9 +100,17 @@ if(STATIC_LINKING)
   target_link_libraries(ast PUBLIC ${clang_libs})
   target_link_libraries(ast PUBLIC ${llvm_libs})
 else()
-  # llvm_config macro comes from the LLVM toolchain and will auto-resolve component
-  # names to library names. USE_SHARED option will tell llvm_config macro to prefer
-  # shared library / DLL on the system over the static libraries
-  llvm_config(ast USE_SHARED bpfcodegen ipo irreader mcjit orcjit)
+  find_package(LLVM QUIET CONFIG)
```

Comment:
Why is this config mode search necessary? Wouldn't it also be necessary for above `llvm_map_components_to_libnames()`, as it's defined in the same source file as `llvm_config()`?

Reply:
Looking again, what happens if you delete this line?

Obviously nothing happens for me on my machine (due to what I'm describing in above comment). What about your setup?

Asking b/c once this goes in nobody else who develops this project will be able to reproduce what you're seeing, so unless we understand why it is the way it is it'll probably regress again eventually.

Reply:
sry for missing this comment. 

> Why is this config mode search necessary?

I believe LLVM was loaded by another CMake file earlier. it's not necessary here. thanks for pointing it out.